bokehjs/src/coffee/models/plots/plot.coffee

```diff
@@ -352,11 +352,16 @@ class PlotView extends Renderer.View
     @update_range(null)
 
   build_levels: () ->
+    renderer_models = @mget("renderers")
+    for tool_model in @mget("tools")
+      synthetic = tool_model.get("synthetic_renderers") ? []
+      renderer_models = renderer_models.concat(synthetic)
```

Comment:
This seems OK, I am not too fond of calling them `synthetic_renderers`. They are just normal renderers that happen to get contributed in a different place (i.e., from a tool).